packages/vite/src/node/plugins/asset.ts

```diff
@@ -414,6 +414,8 @@ const shouldInline = (
 ): boolean => {
   const environment = pluginContext.environment
   const { assetsInlineLimit } = environment.config.build
+  if (noInlineRE.test(id)) return false
+  if (inlineRE.test(id)) return true
   if (environment.config.build.lib) return true
   if (pluginContext.getModuleInfo(id)?.isEntry) return false
   if (forceInline !== undefined) return forceInline
```

Comment:
```suggestion
  if (environment.config.build.lib) return true
  if (pluginContext.getModuleInfo(id)?.isEntry) return false
  if (forceInline !== undefined) return forceInline
  if (noInlineRE.test(id)) return false
  if (inlineRE.test(id)) return true
```
I think these conditions should come before. For example, `<link rel="manifest" href="./manifest.json?some-settings-for-cdn&inline">` will be inlined unexpectedly.

Reply:
I initially had them below too, then I move it back up again. If the user has explicitly specified `?inline` or `?no-inline`, that means they likely want to force that behaviour no matter what, so it doesn't make sense to have certain conditions preventing that. It'd also cause inconsistencies with dev.

`<link rel="manifest" href="./manifest.json?some-settings-for-cdn&inline">` would indeed be inlined, but the user opt themselves to that with `&inline`, so we should respect that.

Reply:
> `<link rel="manifest" href="./manifest.json?some-settings-for-cdn&inline">` would indeed be inlined, but the user opt themselves to that with `&inline`, so we should respect that.

I mixed up `vite-ignore` and `?inline` when thinking about it. Please forget about it 😅